internal/test/e2e/plugins.go

```diff
@@ -405,7 +405,8 @@ var VersionCompatibilityOSSPluginConfigurationTests = []VersionCompatibilityPlug
 	//
 	// DP < 3.0:
 	//   - remove 'http_span_name', 'connect_timeout'
-	//     'send_timeout', 'read_timeout'
+	//     'send_timeout', 'read_timeout',
+	//     'http_response_header_for_traceid'
```

Comment:
Same call out here for the test. We should also create a test to ensure that this field (and eventually others) are going down to a 3.1 data plane which is currently [kong/kong:master-alpine](https://github.com/Kong/koko/blob/6846bca4d60159dc36b72fd645c0c86b028b5026/.github/workflows/integration.yaml#L16) executed during the integration GitHub Action.

Reply:
> We should also create a test to ensure that this field (and eventually others) are going down to a 3.1 data plane

What does this mean? I provide an image tag when testing locally so I am hitting 3.1. Are you saying I also need to make a change to the CICD configuration?

Reply:
`kong/kong:master-alpine` is versioned `3.1.0`, so the CI is already testing the current status of the 3.1 release. We could probably add a test similar to where we use `versions300AndAbove` and make sure new fields are not altered by the compatibility stuff (so going down to a 3.1+ DP). 

@mikefero feel free to add more details in case I misunderstood this.

Reply:
Yep, that is exactly what I was referring to. Since this is for a new version of the Kong Gateway we want to ensure that this new field is being sent in the payload to these newer data planes. Validating these new releases is critical.

Reply:
I've changed the logic to require the field for versions 3.1^ instead of removing the field for versions pre-3.1 [7f84e48](https://github.com/Kong/koko/pull/496/commits/7f84e48d7b13bf09330a71b673c8f1e1968893db).
In the [absence of documentation](https://docs.konghq.com/hub/kong-inc/zipkin/) on the field, I'm assuming we always want to set the value to `X-B3-TraceId` [based on the preceding work.](https://github.com/Kong/kong/pull/9173/files#diff-86d2d554cc8b36c294270788050e859dc26969f5d0495d888e56a27b764d1adfR478).

The reason I went with "remove for versions below 3.1" instead of "ensure presence for 3.1 and above" is because [this field is not required](https://github.com/Kong/kong/pull/9173/files#diff-4b0c5a7443b055f6888c3911c2ca1868b8fbebda607a49b52c9f266ec8f98859R68). Also, this probably should be a warning instead of an error, no?

Reply:
@omegabytes , sorry, but I think there are some misunderstandings with the latest changes. Let me try to clarify:

1. The value of `http_response_header_for_traceid` can be set to anything the user want. This [line](https://github.com/Kong/kong/pull/9173/files#diff-86d2d554cc8b36c294270788050e859dc26969f5d0495d888e56a27b764d1adfR478) you pointed is only based on a test value, which is a simple [plugin configuration](https://github.com/Kong/kong/pull/9173/files#diff-86d2d554cc8b36c294270788050e859dc26969f5d0495d888e56a27b764d1adfR453). 

2. There is no `absence of documentation`, it is just not done yet. The gateway documentation is often done only 1-2 weeks before GA, depending on the workload.

3. We need a ChangeMetadata + test to ensure that if a user configure the zipkin plugin with that field, it is removed  for older DPs. At the moment, this is probably crashing (errors at config push in DP logs). This was OK in you previous version IIRC.

4. We want a test to ensure that if a user configure the zipkin plugin with that field, the plugin configuration is not altered by the version compatibility translator. It should not with your previous version, so it is just a sanity check that new fields of the schema for a version are correctly going down for the new 3.1 release. So that is exactly [this](https://github.com/Kong/koko/pull/496/files#diff-de20d2d66636a4ebbfeaf31f3cd4f7ef469b1fc14c7d2ab47a0e69efb4ba5191R428-R433) (if the plugin in test was configured with that field value)